metricbeat/docs/modules/kubernetes.asciidoc

```diff
@@ -14,11 +14,46 @@ This module fetches metrics from Kubernetes several components:
 - https://kubernetes.io/docs/reference/command-line-tools-reference/kube-scheduler/[scheduler]
 - https://kubernetes.io/docs/reference/command-line-tools-reference/kube-proxy/[proxy]
 
-All metricsets with the `state_` prefix require `hosts` field pointing to kube-state-metrics
-service within the cluster, while the rest should be pointed to kubelet service. Check the
-example configuration on how to do it.
+Some of the previous components are running in each of the Kubernetes nodes (like `kubelet` or `proxy`) while others provide a single cluster-wide endpoint. This is important to determine the optimal configuration and running strategy for the different metricsets included in the module.
 
-The default metricsets are `container`, `node`, `pod`, `system` and `volume`.
+For a complete reference on how to configure and run this module on Kubernetes as part of a `DaemonSet` and a `Deployment`, there's a complete example manifest available in <<running-on-kubernetes,Running Metricbeat on Kubernetes>> document.
+
+[float]
+=== Kubernetes endpoints and metricsets
+
+Kubernetes module is a bit complex as its internal metricsets require access to a wide variety of endpoints.
+
+This section highlights and introduces some groups of metricsets with similar endpoint access needs. For more details on the metricsets see the ****configuration example link*** and ***metricsets reference link**** sections below.
```

Comment:
configuration example link?
metricsets reference link?

Should those be replaced by sth else?

Reply:
@ChrsMark , I'd like to link there to the `Example Configuration` and `Metricsets` sections at the end of the final document. Take a look at the current https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-module-kubernetes.html

How should I add those links?

Reply:
Removing links and resolving conversation :)